### PR TITLE
Minor changes to port cpp

### PR DESCRIPTION
**Pull request by bocchino**
I thought it would be easiest to do these minor changes in a separate PR.

* Consolidate duplicate code in string type generation
* ~Remove system headers -- I don't think they are used~
* Separate system headers from user headers

### CODE REVIEW

File: compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala [hunk 1]
```diff
@@ -157,10 +157,6 @@ case class PortCppWriter (
   }
 
   private def getHppIncludes: CppDoc.Member = {
-    val systemHeaders = List(
-      "cstdio",
-      "cstring",
```
Comment (tiffany1618) [inline compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala:162]: The component implementation classes in F Prime seem to be expecting both of these headers, which aren't included in other files. Some of them use `memset()` from `cstring` or `snprintf()` from `cstdio`, so I think these might be needed here.

**Review by tiffany1618 — COMMENTED** [review]
Looks good! Just one comment

File: compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala (hunk 1 above)
Comment (bocchino) [reply compiler/lib/src/main/scala/codegen/CppWriter/PortCppWriter.scala]: OK, good catch. I think those headers are not needed here, but you're right, the component code may depend on them. Let's fix this later when we revise the component code.

**Comment by bocchino** [comment]
I put the system headers back.

**Review by tiffany1618 — APPROVED** [review]
Looks good!

**Pull request merged**